apis/python/tests/test_dataframe.py

```diff
@@ -783,3 +783,33 @@ def test_result_order(tmp_path):
 
     with pytest.raises(ValueError):
         next(sidf.read(result_order="bogus"))
+
+
+def test_paginated_read(tmp_path):
+    N = 256
+
+    schema = pa.schema(
+        [
+            ("i", pa.int64()),
+            ("soma_joinid", pa.int64()),
+        ]
+    )
+    sidf = soma.DataFrame(uri=tmp_path.as_posix())
+    sidf.create(schema=schema, index_column_names=["i"])
+    data = {
+        "i": list(reversed(range(N))),
+        "soma_joinid": list(range(N)),
+    }
+    sidf.write(pa.Table.from_pydict(data))
+
+    # on defaults, table should read out in one iteration
+    assert sum(x for x in next(sidf.read_as_pandas())["i"].to_list()) == N * (N - 1) / 2
+
+    # setting in platform_config cause it to come out in multiple pages
+    pages = 0
+    total = 0
+    for table in sidf.read_as_pandas(platform_config={"soma.init_buffer_bytes": "64"}):
```

Comment:
@thetorpedodog @johnkerl @gspowley @bkmartinjr I can use some advice here: wiring DataFrame.read() to take platform_config. Our libtiledbsoma SOMAReader receives platform_config as a `Mapping[str,str]`/`map<string,string>`. I've wired it up here as such, and it works for the special libtiledbsoma-specific setting I needed to override...

But I saw in #542 there's some wrapping around `platform_config` where the mapping values have nested structure, and there are some predefined attributes/options that get processed. I couldn't quite tell how/if I should extend that to this use case, to make it consistent and a little higher-level than `Mapping[str,str]`. Any advice on that please? (It's also awkward that I had to write `"64"` as a str here...)

Also noting we may end up revising this in #540 if we end up making the reader more stateful to avoid re-opening the context for each read.

Reply:
The [spec](https://github.com/single-cell-data/SOMA/blob/main/abstract_specification.md#platform-specific-configuration) states that:
> The platform_config parameter is defined as a key–value mapping from strings to configuration data. Each key in the mapping corresponds to the name of a particular SOMA implementation (i.e., the same string returned by the get_storage_engine call).

So I don't know if the C++ code is expecting the _entire_ map, or just the `TileDB`-specific section.  That seems like an internal decision which @gspowley can answer.  Off the top of my head, I can't think of a reason to prefer one other the other - passing down `platform_condig["tliedb"]` seems as good as passing down the entire dict.

Reply:
The C++ code is expecting the `TileDB`-specific section of the `platform_config`. 

It seems like in the python code where a `SOMAReader` object is created, is a good place to pass `platform_config["tiledb"]` to the C++ code.

Reply:
The `tiledb` key of the `platform_config` dictionary is not a raw dict for passing to `tiledb.Config`. The format as defined so far is in the commit message that added it: https://github.com/single-cell-data/TileDB-SOMA/commit/4b4b2645d876bb28789f8ca39a1026361844ada4

In this case I think the thing to do is to add a key like `tiledb_config: {[name: string]: string}` that we will use to construct the config/context itself.